gnome-image-installer/pages/diskimage/gis-diskimage-page.c

```diff
@@ -273,19 +377,30 @@ gis_diskimage_page_populate_model(GisPage *page, gchar *path)
 
   for (file = (gchar*)g_dir_read_name (dir); file != NULL; file = (gchar*)g_dir_read_name (dir))
     {
-      gchar *fullpath = g_build_path ("/", path, file, NULL);
-      if (gis_store_is_unattended() && ufile != NULL)
+      gchar *fullpath = g_build_path ("/", imgpath, file, NULL);
+
+      /* ufile is only set in the unattended case */
+      if (ufile != NULL)
         {
           if (g_str_equal (ufile, file))
-              add_image(store, fullpath);
+            add_image (store, fullpath, NULL);
         }
       else
         {
-          add_image(store, fullpath);
+          add_image (store, fullpath, NULL);
         }
       g_free (fullpath);
     }
 
+  if (is_live &&
+      /* we only use endless.img if there are no other images in the folder */
+      !gtk_tree_model_get_iter_first (GTK_TREE_MODEL (store), &iter) &&
```

Comment:
Not sure whether this check adds much at this point, given we look for the image and signature specifically by name and there is no possible .asc ambiguity.